static/app/components/nav/context.tsx

```diff
@@ -0,0 +1,61 @@
+import {createContext, useContext, useMemo} from 'react';
+
+import {createNavConfig} from 'sentry/components/nav/config';
+import type {NavConfig, NavItemLayout, NavSubmenuItem} from 'sentry/components/nav/utils';
+import {isNavItemActive, isSubmenuItemActive} from 'sentry/components/nav/utils';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export interface NavContext {
+  /** Raw config for entire nav items */
+  config: Readonly<NavConfig>;
+  /** Currently active submenu items, if any */
+  submenu?: Readonly<NavItemLayout<NavSubmenuItem>>;
+}
+
+const NavContext = createContext<NavContext>({config: {main: []}});
+
+export function useNavContext(): NavContext {
+  const navContext = useContext(NavContext);
+  return navContext;
+}
+
+export function NavContextProvider({children}) {
+  const organization = useOrganization();
+  const location = useLocation();
+  /** Raw nav configuration values */
+  const config = useMemo(() => createNavConfig({organization}), [organization]);
+  /**
+   * Active submenu items derived from the nav config and current `location`.
+   * These are returned in a normalized layout format for ease of use.
+   */
+  const submenu = useMemo<NavContext['submenu']>(() => {
+    for (const item of config.main) {
+      if (isNavItemActive(item, location) || isSubmenuItemActive(item, location)) {
+        if (!item.submenu) {
+          return undefined;
+        }
+        if (Array.isArray(item.submenu)) {
+          return {main: item.submenu};
+        }
+        return item.submenu;
```

Comment:
Could we simplify this to? (assuming if statement is doing some amount of nullish coalesce)
```suggestion
        if (Array.isArray(item.submenu)) {
          return {main: item.submenu};
        }
        return item.submenu ?? undefined;
```

Reply:
Addressed in cf98926d685